### PR TITLE
Allow validate_number compare decimal with number

### PR DESCRIPTION
**Pull request by shhavel**
#1661 
#1662

### CODE REVIEW

File: lib/ecto/changeset.ex [hunk 1]
```diff
@@ -1365,6 +1365,7 @@ defmodule Ecto.Changeset do
   end
 
   defp validate_number(field, %Decimal{} = value, message, spec_key, _spec_function, target_value) do
+    target_value = if is_number(target_value), do: Decimal.new(target_value), else: target_value
```
Comment (michalmuskala) [inline lib/ecto/changeset.ex]: It should be safe to always call `Decimal.new/1` here

File: test/ecto/changeset_test.exs [hunk 2]
```diff
@@ -799,6 +799,13 @@ defmodule Ecto.ChangesetTest do
     changeset = changeset(%{"decimal" => Decimal.new(1.5)})
                 |> validate_number(:decimal, greater_than_or_equal_to: Decimal.new(1.5))
     assert changeset.valid?
+
+    changeset = changeset(%{"decimal" => Decimal.new(-1)})
+                |> validate_number(:decimal, greater_than_or_equal_to: -1.5)
+    assert changeset.valid?
+    changeset = changeset(%{"decimal" => Decimal.new(1.5)})
+                |> validate_number(:decimal, greater_than_or_equal_to: 1.5)
+    assert changeset.valid?
```
Comment (michalmuskala) [inline test/ecto/changeset_test.exs]: It's have one test with a float and one with an integer

File: lib/ecto/changeset.ex (hunk 1 above)
Comment (shhavel) [inline lib/ecto/changeset.ex]: @michalmuskala Ok, will remove condition than. Thanks

File: test/ecto/changeset_test.exs (hunk 2 above)
Comment (shhavel) [inline test/ecto/changeset_test.exs]: Will add one test with an integer and one with float instead two tests with float.

**Comment by michalmuskala** [comment]
Thank you! ❤️

**Pull request merged**